### PR TITLE
Fix config profiles quickly get polluted with untouched settings

### LINKED ISSUE #17157: Config profiles quickly get polluted with untouched settings
### Steps to reproduce:
1. Open the configuration profiles dialog
2. Create a new profile, ensure manual activation
3. Save configuration
4. Open NVDA's settings dialog, go to braille settings, then save.
5. Save configuration

### Actual behavior:
Contents of profile after 3:
```ini
schemaVersion = 13
[vision]
	[[NVDAHighlighter]]
		highlightFocus = False
		highlightNavigator = False
		highlightBrowseMode = False
	[[screenCurtain]]
		warnOnLoad = False
		playToggleSounds = True
```

Contents of profile after 5:
```ini
schemaVersion = 13
[vision]
	[[NVDAHighlighter]]
		highlightFocus = False
		highlightNavigator = False
		highlightBrowseMode = False
	[[screenCurtain]]
		warnOnLoad = False
		playToggleSounds = True
[braille]
	translationTable = nl-comp8.utb
	inputTable = nl-comp8.utb
	mode = followCursors
	expandAtCursor = False
	showCursor = True
	cursorBlink = True
	cursorBlinkRate = 500
	cursorShapeFocus = 192
	cursorShapeReview = 128
	showMessages = 1
	messageTimeout = 4
	tetherTo = auto
	readByParagraph = False
	paragraphStartMarker = ""
	speakOnRouting = False
	wordWrap = False
	focusContextPresentation = changedContext
```

### Expected behavior:
Profile should be empty after both step 3 and step 5:

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-33940

#### Windows version:
Windows 11 24h2

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, at least the 2024.3 and 2024.4 cycles have this issues as far as I know

#### If NVDA add-ons are disabled, is your problem still occurring?
Yes

### PR DESCRIPTION
**Pull request by LeonarddeR**
### Link to issue number:
Fixes #17157 
Fixup of https://github.com/nvaccess/nvda/pull/14133
Partially reverts https://github.com/nvaccess/nvda/pull/15074

### Summary of the issue:
When saving configuration profiles, unchanged values are yet saved to a profile.

### Description of user facing changes
Configuration profiles should become less polluted after this change.

### Description of development approach
Restored the behavior of https://github.com/nvaccess/nvda/pull/14133, mostly reverting #15074. There was namely a False assumption in #15074.
@seanbudd wrote:
> This caused various issues including config values not being correctly converted to numbers from strings when validating.

In fact, the validation was not the problem here. Instead, when `__getitem__` was called with `checkValidity` set to False, the unvalidated value was yet saved in the cache. Therefore subsequent calls of `__getitem__` would always return the unvalidated cached value that would always be of the string type.

### Testing strategy:
- [x] Tested STR from #17157. Ensured that the braille and vision settings don't get polluted.
- [x] Tested str from #14133
- [x] Tested str from https://github.com/nvaccess/nvda/issues/14760 (cc @fernando-jose-silva)
- [x] Tested str from https://github.com/nvaccess/nvda/issues/14760#issuecomment-1491684838 (cc @CyrilleB79)

### Known issues with pull request:
None known

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **New Features**
	- Improved handling of configuration settings to prevent redundant storage of identical values in the active configuration profile.
  
- **Bug Fixes**
	- Resolved an issue that caused certain settings to be incorrectly saved, enhancing the integrity of the configuration management process. 

- **Documentation**
	- Updated user documentation to include notes on recent fixes related to configuration settings.

### CODE REVIEW

**Comment by LeonarddeR** [comment]
@CyrilleB79 Would you be able to assist with testing to ensure we don't reintroduce anything from https://github.com/nvaccess/nvda/issues/14760 with this pr?

**Review by seanbudd — COMMENTED** [review]
Is it possible to create a unit test which covers this case in `tests/unit/test_config.py`?

**Comment by LeonarddeR** [comment]
@seanbudd I just created unit tests and checked that the tests fail without the patch in this pr.

File: tests/unit/test_config.py [hunk 1]
```diff
@@ -884,3 +884,29 @@ def test_update_FeatureFlag_defaultValue_fromValueOfDefault(self):
 		self.assertIs(self.testSection["foo"], defaultFlag)
 		self.testSection["foo"] = valueOfDefaultFlag
 		self.assertIs(self.testSection["foo"], valueOfDefaultFlag)
+
+
+class Config_AggregatedSection_polution(unittest.TestCase):
```
Comment (seanbudd) [inline tests/unit/test_config.py:889]:
```suggestion
class Config_AggregatedSection_pollution(unittest.TestCase):
```

**Review by seanbudd — APPROVED** [review]
Perfect, thanks @LeonarddeR

**Pull request merged**